internal/payloads/error_payload.go

```diff
@@ -67,40 +69,117 @@ func PubkeyDuplicateError(ctx context.Context, message string, err error) *Respo
+	var status int
+	var message string
+	switch err {
+	case clients.NotFoundErr:
+		status = 404
+		message = "service returned not found or no data"
+	case clients.UnauthorizedErr:
+		status = 401
+		message = "service returned unauthorized"
+	case clients.ForbiddenErr:
+		status = 403
+		message = "service returned forbidden"
+	case clients.Non2xxResponseErr:
+		status = 500
+		message = "service did not return 2xx"
+	case clients.UnknownAuthenticationTypeErr:
+		status = 500
+		message = "unknown authentication type"
+	case clients.ClientErr:
+		status = 500
+		message = "unknown client error"
+	default:
+		status = 500
+		message = "unknown client error"
+	}
+
 	if logger := ctxval.Logger(ctx); logger != nil {
-		logger.Error().Msg(msg)
+		logger.Error().Msg(message)
 	}
 	return &ResponseError{
 		HTTPStatusCode: status,
-		Message:        msg,
+		Message:        message,
+		TraceId:        ctxval.TraceId(ctx),
+		Error:          err,
+	}
+}
+
+func SourcesError(ctx context.Context, err error) *ResponseError {
+	var status int
+	var message string
+	switch err {
+	case httpClients.MoreThanOneAuthenticationForSourceErr:
+		status = 500
+		message = "more than one authentication"
+	case httpClients.ApplicationNotFoundErr:
+		status = 404
+		message = "application not found"
+	case httpClients.ApplicationTypeNotFoundErr:
+		status = 404
+		message = "unexpected source type"
+	case httpClients.SourceNotFoundErr:
+		status = 404
+		message = "source not found"
+	case httpClients.AuthenticationSourceAssociationErr:
+		status = 500
+		message = "authentication associated to source id not found"
+	case httpClients.AuthenticationForSourcesNotFoundErr:
+		status = 404
+		message = "authentication for source not found"
+	case httpClients.SourcesErr:
+		status = 500
+		message = "unknown sources error"
+	default:
+		status = 500
+		message = "unknown sources error"
+	}
+	if logger := ctxval.Logger(ctx); logger != nil {
+		logger.Error().Msg(message)
+	}
+	return &ResponseError{
+		HTTPStatusCode: status,
+		Message:        message,
+		TraceId:        ctxval.TraceId(ctx),
+		Error:          err,
+	}
+}
+
+func GeneralError(ctx context.Context, message string, err error) *ResponseError {
+	if logger := ctxval.Logger(ctx); logger != nil {
+		logger.Error().Msg(message)
+	}
+	return &ResponseError{
+		HTTPStatusCode: 500,
+		Message:        message,
 		TraceId:        ctxval.TraceId(ctx),
 		Error:          err,
 	}
 }
 
 func NewClientInitializationError(ctx context.Context, message string, err error) *ResponseError {
-	msg := fmt.Sprintf("HTTP client initialization error: %s: %v", message, err)
+	message = fmt.Sprintf("client initialization error: %s", message)
 	if logger := ctxval.Logger(ctx); logger != nil {
-		logger.Error().Msg(msg)
+		logger.Error().Msg(message)
 	}
 	return &ResponseError{
 		HTTPStatusCode: 500,
-		Message:        msg,
+		Message:        message,
 		TraceId:        ctxval.TraceId(ctx),
 		Error:          err,
 	}
 }
```

Comment:
I think you could remove this special error for client initialization and replace it with sources/image builder error defined the same way as the other errors (wrapping a common error).